webapp/channels/src/components/integrations/abstract_oauth_app.tsx

```diff
@@ -292,10 +341,7 @@ export default class AbstractOAuthApp extends React.PureComponent {
                             defaultMessage='Installed OAuth2 Apps'
                         />
                     </Link>
-                    <FormattedMessage
-                        id={headerToRender.id}
-                        defaultMessage={headerToRender.defaultMessage}
-                    />
+                    {this.getHeader()}
```

Comment:
If we can keep the `header` type as `MessageDescriptor` only, we dont need this change